### PR TITLE
Add configuration of the apache reverse proxy

### PR DESCRIPTION
**Pull request by akanshgulati**
Adding config while installing Verdaccio on the root directory as current config fails in that scenario.

### CODE REVIEW

File: docs/reverse-proxy.md [hunk 1]
```diff
@@ -18,6 +20,18 @@ Apache and `mod_proxy` should **not decode/encode slashes** and leave them as th
...
 ````
 
+For installing at root path, `/`, on the server
+````
+<VirtualHost *:80>
+  ServerName registry.url
+  ServerAdmin hello@my-registry.url
+  ProxyPreserveHost On
+  AllowEncodedSlashes NoDecode
+  ProxyPass /  http://127.0.0.1:4873/ nocanon
```
Comment (DanielRuf) [inline docs/reverse-proxy.md]:
```suggestion
  ProxyPass / http://127.0.0.1:4873/ nocanon
```

File: docs/reverse-proxy.md [hunk 2]
```diff
@@ -18,6 +20,18 @@ Apache and `mod_proxy` should **not decode/encode slashes** and leave them as th
 </VirtualHost>
 ````
 
+For installing at root path, `/`, on the server
+````
+<VirtualHost *:80>
+  ServerName registry.url
```
Comment (DanielRuf) [inline docs/reverse-proxy.md]: Let's use example.com or the same hostname for both settings.

**Comment by DanielRuf** [comment]
Can you squash (fixup) all three commits and use the `docs: ...` for the commit (reword)?

This would be great.

**Comment by juanpicado** [comment]
@DanielRuf no worries, in this repo we squash by default as unique option by default

**Comment by DanielRuf** [comment]
> we squash by default

All commits into one? We should not do this as this creates very big diffs / patches and makes it harder in git bisect and blame to find out what commit changed something. Because some commits are just too big.

**Comment by DanielRuf** [comment]
At least not by default.

Squashing all commits by default is not ok for me =)

Can you clarify why we do this? Because some PRs which contain refactors and bugfixes are in one big commit now.

**Comment by akanshgulati** [comment]
> Can you squash (fixup) all three commits and use the `docs: ...` for the commit (reword)?
> 
> This would be great.

@DanielRuf That's one of the feature on Github where we can check the option to squash commits and make a single commit for that. 

I agree squashing is not recommended option for multiple file/line changes but again commit should be made in such a manner to be useful, otherwise multiple commits will also be useless. Even some companies consider as disrespectful for developers for changing a developers log. So, kind a debate-able thing. For my case, I agree it was necessity and it could have been done from Github only while merging, correct me if I am wrong.

**Comment by DanielRuf** [comment]
You can always do `git rebase HEAD~3`, use `fixup` and do a `git push -f` which is ok for PRs.

**Comment by DanielRuf** [comment]
You can always do `git rebase -i HEAD~3`, use `fixup` and do a `git push -f` which is ok for PRs.

Not sure why it should be disrespectful (there is also authored-by).

**Comment by juanpicado** [comment]
@DanielRuf the main reason for have it enable it by default is the `crowdin` integration, which generate thousands of commits on each PR and I merge couple of times by mistake at this repo once by mistake https://github.com/verdaccio/website/graphs/code-frequency 

https://github.com/verdaccio/website/pull/217

Actually, I do not remove never such branch, otherwise will fetch again thousands of files and commits. 

Anyway, I sometimes use the same approach in other repos since not everybody uses git in the cleanest manner unfortunately and merge just add noise rather be useful. Furthermore I try users PR only one single topic so if I must squash then is harmful in order to go backwards and find a root of the bug. As @akanshgulati kinda debate.

**Pull request merged**